src/s2_maybe.erl

```diff
@@ -51,6 +52,19 @@ do_test() ->
        ]).
 -endif.
 
+-spec fmap(fun((A) -> B), maybe(A, _)) -> maybe(B, _).
+%%@doc fmap(F, Maybe) is the result of mapping F over Maybe.
+fmap(F, {ok, Value})     when is_function(F, 1) -> {ok, F(Value)};
+fmap(F, {error, Reason}) when is_function(F, 1) -> {error, Reason}.
+
+-ifdef(TEST).
+fmap_test() ->
+  {ok, 2}               = fmap(fun(X)  -> X + 1            end, {ok, 1}),
+  {ok, {ok, 2}}         = fmap(fun(X)  -> {ok, X + 1}      end, {ok, 1}),
+  {ok, {error, reason}} = fmap(fun(_X) -> {error, reason}  end, {ok, 1}),
+  {error, reason}       = fmap(fun(_X) -> {error, reason2} end, {error, reason}).
```

Comment:
These two tests looks deceptive in my eyes...

Reply:
the second one is not really necessary. the first one looks weird but I think we should keep it. just to make sure that `fmap` doesn't flatten the `maybe`